### PR TITLE
Rollup of 8 pull requests

### PR DESCRIPTION
**Pull request by compiler-errors**
Successful merges:

 - #125667 (Silence follow-up errors directly based on error types and regions)
 - #125717 (Refactor `#[diagnostic::do_not_recommend]` support)
 - #125795 (Improve renaming suggestion for names with leading underscores)
 - #125865 (Fix ICE caused by ignoring EffectVars in type inference)
 - #125953 (Streamline `nested` calls.)
 - #125959 (Reduce `pub` exposure in `rustc_mir_build`)
 - #125967 (Split smir `Const` into `TyConst` and `MirConst`)
 - #125968 (Store the types of `ty::Expr` arguments in the `ty::Expr`)

r? @ghost
@rustbot modify labels: rollup

[Create a similar rollup](https://bors.rust-lang.org/queue/rust?prs=125667,125717,125795,125865,125953,125959,125967,125968)

### CODE REVIEW

**Comment by Kobzol** [comment]
Probably caused by https://github.com/rust-lang/rust/pull/125968. @oli-obk @BoxyUwU Do you think that we can do something about the regression or should we just attribute it to "codegen noise" and leave it be?

**Comment by oli-obk** [comment]
yea, this looks a lot like codegen noise: https://github.com/rust-lang/rust/pull/125968#issuecomment-2149087350

**Comment by Kobzol** [comment]
Ok, marking as triaged.

@rustbot label: +perf-regression-triaged

File: compiler/rustc_borrowck/src/nll.rs [hunk 1]
```diff
@@ -144,6 +144,16 @@ pub(crate) fn compute_regions<'cx, 'tcx>(
...
+        // Suppress unhelpful extra errors in `infer_opaque_types` by clearing out all
+        // outlives bounds that we may end up checking.
+        outlives_constraints = Default::default();
+        member_constraints = Default::default();
+
+        // Also taint the entire scope.
+        infcx.set_tainted_by_errors(guar);
+    }
+
     let mut regioncx = RegionInferenceContext::new(
```
Comment (Allmeekadoo) [inline compiler/rustc_borrowck/src/nll.rs:157]: Hello

**Review by Allmeekadoo — COMMENTED** [review]
Hello

**Pull request merged**